### PR TITLE
[#44] Fix to pass the channel if it exists

### LINKED ISSUE #44: slacker.client/receive-message seems passes no arguments
I created a test branch of my slacker sample repository.

https://github.com/supermomonga/sample-slacker-pingpong/blob/1cc8a047b670c5ba1a66fd8afe3f4570e50c4dd8/src/sample_slacker_pingpong/core.clj#L16-L20

It tries to print passed args when some events happened (hello, message...).

I expect [some arguments will be passed](https://github.com/emiln/slacker/blob/841c542c65ae8d2a2191022bab4a4dcf5102def5/src/slacker/client.clj#L110) but it seems always passed no argument.

Here is the log when I run a bot with `lein run` command and tried some actions in my personal slack team's channel which have two members, bot and me.

```
hello args: nil
message args: nil
pin-added args: nil
message args: nil
pin-removed args: nil
```

### PR DESCRIPTION
**Pull request by supermomonga**
Accepting this closes #44.

### CODE REVIEW

**Comment by emiln** [comment]
That's nicely spotted, @supermomonga! It seems I never updated the number of arguments sent to `publisher` after I changed the arity of `handle`. You are completely right that this is in fact where the bug lies.

However, due to an annoying clash of naming conventions, the channel that `handle` expects is actually a `core.async` channel rather than a Slack channel. The idea is that if a non-nil channel is supplied, and if the handler function produces a result, then this result can be communicated back. This allows flow like the following:

``` clojure
(handle :add-up +)
(<!! (emit-with-feedback! :add-up 1 2 3 4 5))
```

where line 2 blocks until the value `15` is communicated back over the channel.

What all this boils down to is that the `receive-message` function should simple pass `nil` like so:

``` clojure
  (let [topic (cond (:type msg)     (:type msg)
                    (:reply_to msg) "reply"
                    :else           "unknown")]
    (go (>! publisher [(string->keyword topic) nil msg]))))
```

This will prevent the handler from attempting to pass anything back through any feedback channel. There are a few scenarios where passing a String containing a channel ID will cause trouble.

**The TL;DR**:

Thanks a lot for finding the bug. If you're up for changing the PR slightly as outlined above I'll approve it and you'll have fixed the bug.

**Comment by supermomonga** [comment]
Thank you for your review and explanation.
Yes, as you said, I confused slack channel with core.async's channel. what a embarrassing mistake... I should read whole source code again.

Anyway, I'll fix this PR immediately :smiley:

**Comment by emiln** [comment]
Excellent. :-)
